flow/view_holder.cc

```diff
@@ -0,0 +1,121 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <flutter/flow/view_holder.h>
+
+#include <flutter/fml/make_copyable.h>
+#include <fuchsia/ui/gfx/cpp/fidl.h>
+
+namespace flow {
+
+ViewHolder::ViewHolder(
+    fml::RefPtr<zircon::dart::Handle> view_holder_token_handle,
+    std::function<void(scenic::ResourceId)> on_bind_callback,
+    std::function<void(scenic::ResourceId)> on_unbind_callback)
+    : on_bind_callback_(std::move(on_bind_callback)),
+      on_unbind_callback_(std::move(on_unbind_callback)) {
+  view_holder_token_.value.reset(view_holder_token_handle->ReleaseHandle());
+}
+
+ViewHolder::~ViewHolder() {
+  // Ensure that we properly released the resource.
+  FML_DCHECK(!view_holder_);
+  FML_DCHECK(scene_update_context_ == nullptr);
+}
+
+void ViewHolder::Bind(SceneUpdateContext& context,
+                      scenic::ContainerNode& container,
+                      const SkPoint& offset,
+                      bool hit_testable) {
+  FML_LOG(ERROR) << "ViewHolder::Bind";
+  if (view_holder_token_.value) {
+    FML_LOG(ERROR) << "CREATE NODES";
+    // Happens first time we bind.
+    // The Resource ID message MUST be posted before the resource is created, or
+    // race conditions could occur!
+    scenic::ResourceId view_holder_id = container.session()->next_resource_id();
+    on_bind_callback_(view_holder_id);
+
+    // Create resources.  This might cause an immediate session flush, so this
+    // must be done after registering the view_holder_id via the callback.
+    view_holder_.reset(new scenic::ViewHolder(
+        container.session(), std::move(view_holder_token_.value),
+        "Flutter SceneHost"));
+    transform_node_.reset(new scenic::EntityNode(container.session()));
+    transform_node_->Attach(*view_holder_);
+
+    scene_update_context_ = &context;
+  }
+
+  if (view_holder_) {
+    FML_LOG(ERROR) << "ATTACH NODES";
+    transform_node_->SetTranslation(offset.x(), offset.y(), 0.f);
```

Comment:
We want to add a fixed elevation to avoid z-fighting with children.  The correct value is TBD, but 1.0 is probably a good starting point.

Reply:
I added -0.1, that is what we have been using everywhere else